storage/tests/system.py

```diff
@@ -360,6 +360,15 @@ def test_list_files(self):
         self.assertEqual(sorted(blob.name for blob in all_blobs),
                          sorted(self.FILENAMES))
 
+    @unittest.skipUnless(USER_PROJECT, 'USER_PROJECT not set in environment.')
+    @RetryErrors(unittest.TestCase.failureException)
+    def test_list_files_with_user_project(self):
+        with_user_project = Config.CLIENT.bucket(
+            self.bucket.name, user_project=USER_PROJECT)
+        all_blobs = list(with_user_project.list_blobs())
```

Comment:
This isn't very useful unless we add blobs first?

Reply:
Done in `setupClass`.

Reply:
Ah yes, I recall that now.

Reply:
Any remaining issues?

Reply:
Nope LGTM